### ISSUE TITLE
`flow project deploy` broken on windows

### ISSUE THREAD

**Issue by jacob-tucker** [onflow/flow-cli#1183]
When I deploy my contracts in flow.json on linux/mac, it works totally fine. But windows seems to be broken.

![image](https://github.com/onflow/flow-cli/assets/15198786/e417b525-8b51-4990-944c-6922a1a4a968)
Something to do with not reading the imports in contracts correctly. Maybe due to wrongful file paths

**Comment by sideninja** [comment]
@chasefleming this seems like a regression. Can you look into it please?

**Comment by jribbink** [comment]
I did some debugging here & it appears that this has to do with OS-specific path formatting.  Essentially, in some places in the code a Windows-format URL (backslashes) is generated and eventually this is used with functions from the `path` module (not `filepath`).

This surfaces here https://github.com/onflow/flow-cli/blob/master/flowkit/project/deployment.go#L141-L142 where `contract.location` is a Windows path, but `absolutePath` expects UNIX-style paths.  The absolute path does not form properly because the backslashes are not supported.

I did a lot of manual testing and ultimately determined a few things
1. Prior to `v0.47.0`, **UNIX-style** paths listed in the deployments section of flow.json _could be_ resolved as imports for other deployments.  Something broke in this release that broke this functionality
2. **Windows-style** paths (unless a very very long time ago) have never appeared to be supported by the Flow CLI, even on Windows.
3. `flow dev` and potentially other commands generate Windows-style paths and fill these into the deployments section automatically.  This is problematic because of point no. 2.  At no point have Windows-style paths been supported, so fixing the regression in no. 1 alone would not be sufficient completely fix this bug.

I think the solution here is to enforce either:
1. That all paths in flow.json are in UNIX-format, `flow dev` & other commands write in UNIX format, and we fix whatever broke UNIX-format in Windows originally
2. That both UNIX & Windows format paths work equivocally in the CLI regardless of OS.
